File: src/MyTeam/Models/Domain/Club.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MyTeam.Models.Domain
{
    public class Club : Entity
    {
        [Required]
        public string ClubIdentifier { get; set; }
        [Required]
        public string ShortName { get; set; }
        [Required]
        public string Name { get; set; }
        [Required]
        public string Logo { get; set; }
        public string Favicon { get; set; }
        public virtual List<Team> Teams { get; set; }
        public virtual List<Member> Players { get; set; } 
    }
}

File: src/client/src/domain/Club.fs
namespace Shared.Domain

open System

type TeamId = Guid

type LeagueType =
    | Syver
    | Ellever

type Team =
    { Id: TeamId
      ShortName: string
      Name: string
      LeagueType: LeagueType }

type ClubIdentifier = ClubIdentifier of string
type ClubId = ClubId of Guid

type Club =
    { Id: ClubId
      ClubId: string
      ShortName: string
      Name: string
      Teams: Team list
      Favicon: string
      Logo: string }
